django/db/models/fields/related.py

```diff
@@ -807,6 +807,8 @@ def __init__(self, to, on_delete, related_name=None, related_query_name=None,
             on_delete=on_delete,
         )
         kwargs.setdefault('db_index', True)
+        if parent_link:
+            kwargs['returning'] = True
```

Comment:
Could be made a property that returns `super().returning or self.remote_field.parent_link`.

Reply:
I'll move it.